### PR TITLE
Feature #13 uploading images for posts

### PR DESCRIPTION
**Pull request by kiyaGu**
I have also included in this PR "#14 ability to put a link related to a post" as it is just including one attribute besides the image url. So, I have done them together

### CODE REVIEW

File: api/src/controllers/ImageController.js [hunk 1]
```diff
@@ -108,6 +121,33 @@ class ImageController extends Controller {
       }
     });
   }
+
+  //post
+  imagePost_Post_Handler(request, reply) {
```
Comment (kkarimi) [inline api/src/controllers/ImageController.js]: Underscores are not idiomatic JS names

File: api/src/controllers/ImageController.js [hunk 2]
```diff
@@ -108,6 +121,33 @@ class ImageController extends Controller {
...
+
+    let newImageKey = Hasher.encode(request.auth.credentials.id, Date.now());
+    let s3 = new AWS.S3();
+
+    uploadImage(s3, body, `post/${newImageKey}`, (error, data) => {
+      if (error) {
+        reply({ msg: error }).code(400);
+      } else {
+        reply({
+          imageSource: data.Location
```
Comment (kkarimi) [inline api/src/controllers/ImageController.js]: Not important, but capital letters are not idiomatic JS key names

File: web/src/components/Post.js [hunk 3]
```diff
@@ -27,9 +28,15 @@ export default class Post extends Component {
     useCurrentLocationTooltipOpen: false,
     postTypeAskChecked: false,
     postTypeOfferChecked: false,
+    //default image
+    imageSource:
+      "https://humankind-assets.s3.eu-west-1.amazonaws.com/post/gr8QHk31k2Raa",
```
Comment (kkarimi) [inline web/src/components/Post.js]: Not important here but this could also be a constant with a meaningful name

File: web/src/components/Post.js [hunk 4]
```diff
@@ -121,24 +134,23 @@ export default class Post extends Component {
...
   };
 
   handleSubmitRequest = e => {
     e.persist();
-    let buttonText = e.target.textContent.trim();
-    //implement post
-    //remove the redirect state when constructing the body of the request
 
+    let buttonText = e.target.textContent.trim();
     let {
```
Comment (kkarimi) [inline web/src/components/Post.js]: this is probably a `const`

File: web/src/components/Post.js [hunk 5]
```diff
@@ -203,6 +221,7 @@ export default class Post extends Component {
   };
 
   getLocation() {
+    const TIME_OUT_SECOND = 6000;
```
Comment (kkarimi) [inline web/src/components/Post.js]: nice!

File: api/src/controllers/ImageController.js (hunk 1 above)
Comment (kiyaGu) [reply api/src/controllers/ImageController.js]: Yeah I thought about that but did it like that as imagePostPostHandler might create confusion... will change it

File: api/src/controllers/ImageController.js (hunk 2 above)
Comment (kiyaGu) [reply api/src/controllers/ImageController.js]: That is not mine it comes from AWS

File: web/src/components/Post.js (hunk 3 above)
Comment (kiyaGu) [reply web/src/components/Post.js]: good point

File: web/src/components/Post.js (hunk 4 above)
Comment (kiyaGu) [reply web/src/components/Post.js]: done that

File: web/src/components/Post.js (hunk 3 above)
Comment (kkarimi) [reply web/src/components/Post.js]: define the const above the class definition?

File: api/src/controllers/ImageController.js (hunk 2 above)
Comment (kkarimi) [reply api/src/controllers/ImageController.js]: ah!

**Pull request merged**